src/Text/Pandoc/Readers/Vimwiki.hs

```diff
@@ -0,0 +1,398 @@
+{-
+  Copyright (C) 2017 Yuchen Pei <me@ypei.me>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+-}
+
+{- |
+   Module      : Text.Pandoc.Readers.Vimwiki
+   Copyright   : Copyright (C) 2017 Yuchen Pei
+   License     : GNU GPL, version 2 or above
+
+   Maintainer  : Yuchen Pei <me@ypei.me>
+   Stability   : alpha
+   Portability : portable
+
+Conversion of vimwiki text to 'Pandoc' document.
+-}
+{--
+ progress:
+* block parsers:
+    * [X] header
+    * [X] hrule
+    * [X] comment
+    * [X] blockquote -- currently only accepting four spaces rather than a mix of spaces and tabs
+    * [X] preformatted -- need to implement preformatted with attributes
+    * [X] displaymath - a bit buggy
+    * [X] bulletlist / orderedlist - a bit buggy with nested lists -- currently not calculating mixed tabs and spaces indentations.
+        * [ ] orderedlist with 1., i., a) etc identification.
+        * [ ] todo lists -- see https://github.com/LarsEKrueger/pandoc-vimwiki
+    * [X] table
+        * [ ] centered table -- there does not seem to be a table builder in Pandoc.Builder that accepts attributes
+        * [O] colspan and rowspan -- pandoc limitation, see issue #1024
+    * [X] paragraph
+    * [ ] definition list
+* inline parsers:
+    * [X] bareURL
+    * [X] strong
+    * [X] emph
+    * [X] strikeout
+    * [X] code
+    * [X] link
+        * [ ] with thumbnails -- pandoc limitation? can't find builder of link with thumbnails
+    * [X] image -- to implement images with attributes - same as in preformatted
+    * [X] inline math - a bit buggy
+    * [X] tag
+    * [ ] sub- and super-scripts
+* misc:
+    * [X] `TODO:` mark
+    * [ ] placeholders: %title and %date -> metadata, %template -> template
+--}
+
+module Text.Pandoc.Readers.Vimwiki ( readVimwiki
+                                 ) where
+import Control.Monad.Except (throwError)
+import Control.Monad (guard)
+import Data.Default 
+import Data.Functor.Identity
```

Comment:
Do you actually use this?
It's only in base starting with 4.8, which is why some of the travis builds failed.
But I don't see you using `runIdentity` anywhere in the code.